cognite/powerops/client/data_classes/shop_results_compare.py

```diff
@@ -38,9 +47,221 @@ def plot_time_series(
         ax.legend()
         plt.show()
 
-    def yaml_difference(
+    def yaml_deep_diff(
         self,
-        yaml_1: ShopYamlFile,
-        yaml_2: ShopYamlFile,
+        post_run_yaml_1: ShopYamlFile,
+        post_run_yaml_2: ShopYamlFile,
+        **deepdiff_kwargs,
+    ) -> DeepDiff:
+        """DeepDiff wrapper for ShopYamlFile objects."""
+        return DeepDiff(
+            post_run_yaml_1.data,
+            post_run_yaml_2.data,
+            ignore_type_in_groups=[(int, float)],
+            **deepdiff_kwargs,
+        )
+
+    def yaml_difference_md(
+        self,
+        post_run_yaml_a: Optional[ShopYamlFile] = None,
+        post_run_yaml_b: Optional[ShopYamlFile] = None,
+        name_a: str = "Result A",
+        name_b: str = "Result B",
+        **deepdiff_kwargs,
+    ):
+        """
+        Returns a markdown string of the difference between two post run yaml files
+        In a notebook, this can be displayed using
+        ```
+        from IPython.display import Markdown
+
+        Markdown(yaml_difference_md(...))
+        ```
+        """
+        if post_run_yaml_a is None or post_run_yaml_b is None:
+            logger.error("Must provide two post run yaml files ")
+            return ""
+
+        yaml_deep_diff = self.yaml_deep_diff(post_run_yaml_a, post_run_yaml_b)
+        builder = YamlDiffMDBuilder(
+            yaml_deep_diff, post_run_yaml_a.data, post_run_yaml_b.data, name_a, name_b, **deepdiff_kwargs
+        )
+
+        output_string = f"# Changes from {name_a} to {name_b}"
+        output_string += "\n\n"
+        output_string += builder.build_md()
+        return output_string
+
+
+class YamlDiffMDBuilder:
```

Comment:
Not sure about this, but maybe make this Private. We are not exposing this do end users. 

```suggestion
class _YamlDiffMDBuilder:
```